File: DSA/Beginner/02-Arrays/Dynamic Arrays/DynamicArray.java
public class DynamicArray {
    int capacity; 
    int length; 
    int[] arr;

    public DynamicArray() {
        capacity = 2; 
        length = 0;
        arr = new int[2];
    }

    public void pushback(int n) {
        if (length == capacity) {
            this.resize();
        }
        arr[length] = n;
        length++;
    }

    private void resize() {
        capacity = 2 * capacity;
        int[] newArr = new int[capacity];
        for (int i = 0; i < length; i++) {
            newArr[i] = arr[i];
        }
        arr = newArr; 
    }

    public void popback() {
        if (length > 0) {
            length--;
        }
    }

    public int get(int i) {
        if (i < length) {
            return arr[i];
        }
        return -1; 
    }

    public void insert(int i, int n) {
        if (i < length) {
            arr[i] = n;
        }
    }

    public void print() {
        for (int i = 0; i < length; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }

}

File: DSA/Beginner/02-Arrays/Dynamic Arrays/DynamicArray.js
class DynamicArray {
    constructor() {
        this.capacity = 2;
        this.length = 0;
        this.arr = new Array(2);
    }

    pushback(n) {
        if (this.length == this.capacity) {
            this.resize();
        }
        this.arr[this.length] = n;
        this.length++;
    }

    resize() {
        this.capacity = 2 * this.capacity;
        const newArr = new Array(this.capacity);
        for (let i = 0; i < this.length; i++) {
            newArr[i] = this.arr[i];
        }
        this.arr = newArr;
    }

    popback() {
        if (this.length > 0) {
            this.length--;
        }
    }

    get(i) {
        if (i < this.length) {
            return this.arr[i];
        }
        return undefined;
    }

    insert(i, n) {
        if (i < this.length) {
            this.arr[i] = n;
        }
    }

    print() {
        let s = ""; 
        for (let i = 0; i<this.length; i++) {
            s += arr[i] + " ";
        }
        console.log(s);
    }
}